NullabilityInference/EdgeBuildingOperationVisitor.cs

```diff
@@ -1665,6 +1665,7 @@ public override TypeWithNode VisitDelegateCreation(IDelegateCreationOperation op
                     }
                     break;
                 case IMethodReferenceOperation methodReference:
+                    if (methodReference.Instance != null && methodReference.Method.IsExtensionMethod) delegateParameters.Insert(0, methodReference.Instance.Accept(this, EdgeBuildingContext.Normal));
```

Comment:
`HandleMethodGroup` itself also visits `methodReference.Instance` via `GetReceiverType`.
But that's wrong for extension methods, as there the instance is unrelated to the class defining the extension method. (direct invocations don't have this issue as there Roslyn already sets `Instance==null` for extension method calls)

So I guess this logic should replace the `GetReceiverType` call. We can then pass `null` as receiverType into `ClassTypeArgumentsForMemberAccess`, since extension methods can't be defined in generic classes.